here-naksha-lib-jbon/src/commonMain/kotlin/JbPath.kt

```diff
@@ -8,20 +8,93 @@ import kotlin.js.JsExport
+                return valueReader.readBoolean() ?: alternative
+            }
+            return alternative
+        }
+
+        fun getInt32(binary: ByteArray, path: String, alternative: Int? = null): Int? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null && valueReader.isInt32()) {
+                return valueReader.readInt32()
+            }
+            return alternative
+        }
+
+        fun getInt64(binary: ByteArray, path: String, alternative: Int? = null): Int? {
+            TODO("int64 has to be implemented")
+        }
+
+        fun getFloat32(binary: ByteArray, path: String, alternative: Float? = null): Float? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null && valueReader.isFloat32()) {
+                return valueReader.readFloat32()
+            }
+            return alternative
+        }
+
+        fun getDouble(binary: ByteArray, path: String, alternative: Double? = null): Double? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null) {
+                if (valueReader.isFloat32()) {
+                    return valueReader.readFloat32().toDouble()
+                } else if (valueReader.isFloat64()) {
+                    return valueReader.readDouble()
+                }
+            }
+            return alternative
+        }
+
+        fun getString(binary: ByteArray, path: String, alternative: String? = null): String? {
+            val valueReader = readElement(binary, path)
+            if (valueReader != null && valueReader.isString()) {
+                return valueReader.readString()
+            }
             return alternative
         }
+
+        private fun readElement(binary: ByteArray, path: String): JbReader? {
+            val session = JbSession.get()
+            if (!bytes.contentEquals(binary)) {
+                bytes = binary
+                view = session.newDataView(bytes)
+                reader = JbReader()
+                reader.mapView(view, 0)
+                feature = JbFeature().mapReader(reader)
+                jmap = JbMap()
+                jmap.mapReader(feature.reader)
+            }
+
+            if (feature.reader.isMap()) {
+                val pathElements = path.split(".")
+                return goToElement(jmap, pathElements)
+            }
+            return null
+        }
+
+        fun goToElement(map: JbMap, path: List<String>): JbReader? {
+            if (path.isEmpty()) {
+                return null
+            }
+            val currentLevelKey = path[0]
+            if (map.selectKey(currentLevelKey) && map.ok()) {
+                return if (map.value().isMap()) {
+                    val newMap = JbMap().mapReader(map.value())
+                    goToElement(newMap, path.drop(1))
+                } else {
+                    map.value()
+                }
+            }
+            return null;
+        }
```

Comment:
there is no get Map, Collection, ByteArray yet,
and reader is not using Dictionary